lib/graphql/batch.rb

```diff
@@ -1,6 +1,48 @@
 require "graphql"
+
+module GraphQL
+  module Batch
+    module_function
+
+    def execute(container)
+      unless container.is_a?(QueryContainer)
+        return container
+      end
+      if container.completed?
+        raise container.error if container.error
+        return container.result
+      end
+
+      batched_queries = Hash.new{ |hash, key| hash[key] = [] }
+      register_queries = lambda do |query_container|
```

Comment:
Does it make sense to define an `Executor` class which has `register_queries` as a method and `batched_queries` as an instance var? I think it would break up the complexity a bit.